dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dimension/DefaultDimensionService.java

```diff
@@ -369,36 +367,7 @@ public DimensionalItemObject getDataDimensionalItemObject( DimensionalItemId ite
     @Override
     public Set<DimensionalItemObject> getDataDimensionalItemObjects( Set<DimensionalItemId> itemIds )
     {
```

Comment:
I am sure it is, but is it intentional to remove such a large code block?

Reply:
Yes, you're right it's intentional. By rethinking the approach to allowing any number of .periodOffset values, all this special-case logic that handles 0 or 1 values is not needed! (See "How the code was changed" above: "The complex filtering...") The cost of fixing the function and eliminating all this special-case logic is that we need some kind of object copying as discussed above.

Reply:
Sounds good.